### ISSUE TITLE
Fields must have labels or titles not accepting aria-label

### ISSUE THREAD

**Issue by andymsuk** [bbc/bbc-a11y#245]
## Summary

The following test fails unexpectedly for buttons with an `aria-label`:
Form labels: Fields must have labels or titles

## Expected Behaviour

Given a page with the following markup:
```<button class="arrows__chevron" aria-label="Scroll carousel right"></button>```

Then the test "Fields must have labels or titles" passes, as it does with Google Lighthouse - https://developers.google.com/web/tools/lighthouse/audits/form-labels

## Current Behaviour

The test fails as the button does not contain text, have an associated `<label>` or have a `title` attribute

## Possible Solution

As per Google Lighthouse, allow `aria-label` and `aria-labelledby`

**Comment by EmmaJP** [comment]
https://www.w3.org/WAI/tutorials/forms/labels/#labelling-buttons

**Comment by andymsuk** [comment]
The first part of your response, @EmmaJP, suggests that we should fix bbc-a11y as I suggested.
The last paragraph suggests we shouldn't change bbc-a11y and that we should use text within the button instead.
Please can you clarify?

**Comment by ermi-ltd** [comment]
Hi,

I’d agree with Andy and would support updating the code.

There are better ways than aria-label and aria-labelledby but they are both valid

**Comment by EmmaJP** [comment]
An aria-label is accessible ... so it should probably pass a bbc-a11y check for a label.

However, if there is a standard way to provide that label with HTML, such as with an alt attribute on an image input, then I would encourage doing that over aria. 

Could bbc-a11y determine and warn if another method should have been used? That's getting rather complex!!

**Comment by andymsuk** [comment]
If someone wants to make a list of elements and attributes/values and which should warn and which should error, I can look at doing that. Otherwise, I can just add aria-label and aria-labelledby to the list of allowed ways of labelling.

**Comment by EmmaJP** [comment]
Perhaps do the latter, and see what other issues come up.

**Comment by oluoluoxenfree** [comment]
Auto-wcag doesn't seem to have an explicit rule about it, the rules around labelling aren't very strict so long as a label exists unless I'm missing something: https://auto-wcag.github.io/auto-wcag/rules/SC2-4-6-descriptive-labels.html

**Comment by EmmaJP** [comment]
The latest version of auto-wcag, now ACT Rules, is at https://github.com/bbc/bbc-a11y/pull/302 and allows for a button with no text and an aria-label.

I think so long as the aria-label also serves as an alternative for something visible, such as a suitable image or icon, bbc-a11y should pass this - although that may require a manual check to confirm. I guess the code could also check for an applied background image or similar if an aria-label is used and there is no other text.

Examples:

aria-label provides text alternative for a background image applied via a CSS style:
 `<button class="arrows__chevron" aria-label="Scroll carousel right"></button>`

aria-label provides additional context for shorter visible text with a layout context:
`<button id="S1" class="share-btn" aria-labelledby="S1 itemTitle">Share</button>`

aria-label is not necessary for an image button, though it could still pass:
 `<input type="button" src="./right-chevron.jpg" alt="Scroll carousel right"/>`
 `<input type="button" src="./right-chevron.jpg" aria-label="Scroll carousel right"/>`

**Comment by greenc05** [comment]
In the background image example, when CSS is disabled/not loaded, the user wouldn't see the aria-label, is this acceptable?

**Comment by EmmaJP** [comment]
Good question!
